packages/devextreme/js/__internal/grids/grid_core/views/m_columns_view.ts

```diff
@@ -792,11 +792,23 @@ export class ColumnsView extends ColumnStateMixin(modules.View) {
     }
   }
 
-  protected _updateCells($rowElement, $newRowElement, columnIndices) {
+  protected _updateCells($rowElement, $newRowElement, columnIndices, options?) {
     const $cells = $rowElement.children();
     const $newCells = $newRowElement.children();
     const highlightChanges = this.option('highlightChanges');
     const cellUpdatedClass = this.addWidgetPrefix(CELL_UPDATED_ANIMATION_CLASS);
+    const {
+      cells: cellElements,
+      node,
+      rowType,
+    } = options || {};
+
+    if (cellElements && node?.hasChildren) {
+      cellElements.forEach((cell) => {
+        cell.cellElement = cell.cellElement instanceof HTMLElement ? $(cell.cellElement) : cell.cellElement;
+        this.setAria('expanded', cell.isExpanded, cell.cellElement);
+      });
+    }
```

Comment:
The advantage of writing the code to the appropriate place is that it might already contain everything needed for it to work. In your case, the set of cells is already stored in the variable:
`const $cells = $rowElement.children();`
So you do not need to destructure options to cells: cellElements. Also rowType variable is not used.
Row's cells have the same aria-expanded attribute as the parent row. So you do not need to use cells[i].isExpanded.

After the simplifications I got the folowing code. Please check:
```
protected _updateCells($rowElement, $newRowElement, columnIndices, options?) {
    const that = this;
    const $cells = $rowElement.children();
    const $newCells = $newRowElement.children();
    const highlightChanges = this.option('highlightChanges');
    const cellUpdatedClass = this.addWidgetPrefix(CELL_UPDATED_ANIMATION_CLASS);

    if (options?.node?.hasChildren) {
      $cells.each(function () {
        that.setAria('expanded', options.isExpanded, $(this));
      });
    }
...

```

Reply:
This is great. Thanks for this!!